e2e/command_test.go

```diff
@@ -159,6 +159,26 @@ func TestE2E_EnableTaskCommand(t *testing.T) {
 			status, ok := taskStatuses[disabledTaskName]
 			require.True(t, ok)
 			assert.Equal(t, tc.expectEnabled, status.Enabled)
+
+			if !tc.expectEnabled {
+				// only check for events if we expect the task to be enabled
+				return
+			}
+
+			// check that there was an initial event
+			eventCountBase := 1
+			eventCountNow := eventCount(t, disabledTaskName, cts.Port())
+			require.Equal(t, eventCountBase, eventCountNow,
+				"event count did not increment once. task was not triggered as expected")
+
+			// make a change in Consul and confirm a new event is triggered
+			now := time.Now()
+			service := testutil.TestService{ID: "api-1", Name: "api"}
+			testutils.RegisterConsulService(t, srv, service, defaultWaitForRegistration)
+			api.WaitForEvent(t, cts, disabledTaskName, now, defaultWaitForEvent)
+			eventCountNow = eventCount(t, disabledTaskName, cts.Port())
+			require.Equal(t, eventCountBase+1, eventCountNow,
+				"event count did not increment once. task was not triggered as expected")
```

Comment:
Don't you need to also check for generated files here as well? To verify the task actually runs?

Reply:
We could! The reason I didn't  is because I found out that even before the fix, there wasn't a situation where the event was created but the task didn't run. I know you had mentioned that when we last discussed this bug, but I believe the confusion is that you were looking at the reenable test, which starts with an enabled task.

But no real reason to leave it out, there's always a chance it could be a bug in the future.

Reply:
Added the check for services!